src/include/firebird/FirebirdInterface.idl

```diff
@@ -1557,16 +1558,39 @@ interface Int128 : Versioned
 
 interface ReplicatedRecord : Versioned
 {
+	// Return count of fields
+	uint getCount();
+	// Return name of field. NULL is returned on error.
+	const string getField(uint index);
+	// Return dtype_* constants or dtype_unknown if field not exists.
+	uint getType(uint index);
+	// Return subtype if applicable
+	uint getSubType(uint index);
+	// Return false if field value is not NULL, true otherwise.
+	boolean isNull(uint index);
+	// Return scale fo the field. Behavior is undefined for non-numeric fields.
+	uint getScale(uint index);
+	// Return maximum length of data in bytes. 0 is returned on error.
+	uint getLength(uint index);
+	// Return character set without collation part for text or CLOB fields. Behavior is undefined for the rest.
+	uint getCharSet(uint index);
+	// Return pointer to data. NULL is returned on error.
+	const void* getData(uint index);
```

Comment:
With the ReplicatedField interface, getData() may return NULL if the value is NULL. isNull() method becomes unnecessary.